backend/assignments/1/src/main/java/org/example/Coins.java

```diff
@@ -0,0 +1,79 @@
+package org.example;
+
+public class Coins {
+    public static final LoggingSystem ls = new LoggingSystem();
+    private int rank;
+    private String name;
+    private String symbol;
+    private double price;
+    private long circulationSupply;
+
+    public Coins(){
+
+    }
+
+    public Coins(int i, String bitcoin, String btc, double v, long l) {
+        this.rank = i;
+        this.name = bitcoin;
+        this.symbol = btc;
+        this.price = v;
+        this.circulationSupply = l;
+    }
+
+
+
+    public  int getRank() {
+        return rank;
+    }
+
+    public  void setRank(int rank) {
+        this.rank = rank;
+    }
+
+    public  String getCoinName() {
+        return name;
+    }
+
+    public  void setCoinName(String name) {
+        this.name = name;
+    }
+
+    public String getCoinSymbol(){
+        return symbol;
+    }
+    public  void setSymbol(String symbol) {
+        this.symbol = symbol;
+    }
+
+    // sync updates in coin....
+    public synchronized double getPrice() {
```

Comment:
What if user1 needs to work on Coin1 and user2 on coin2.
But user2 will still be in wait state untill the method reopoens lock.
Can the locking be in a better place?